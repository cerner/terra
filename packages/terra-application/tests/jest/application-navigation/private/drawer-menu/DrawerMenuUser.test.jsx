import React from 'react';
import DrawerMenuUser from '../../../../../src/application-navigation/private/drawer-menu/_DrawerMenuUser';

const userActionConfig = {
  text: 'Edit Photo',
  userActionCallback: () => jest.fn(),
};

describe('DrawerMenuUser', () => {
  it('should render default element', () => {
    const shallowComponent = shallow(
      <DrawerMenuUser
        userConfig={{
          name: 'user-name',
          detail: 'user-detail',
          initials: 'user-initials',
          imageSrc: 'user-src',
        }}
        userActionConfig={userActionConfig}
      />,
    );

    expect(shallowComponent).toMatchSnapshot();
  });

  it('should render with small variant', () => {
    const shallowComponent = shallow(
      <DrawerMenuUser
        userConfig={{
          name: 'user-name',
          detail: 'user-detail',
          initials: 'user-initials',
          imageSrc: 'user-src',
        }}
        variant="small"
        userActionConfig={userActionConfig}
      />,
    );

    expect(shallowComponent).toMatchSnapshot();
  });

  it('should render with large variant', () => {
    const shallowComponent = shallow(
      <DrawerMenuUser
        userConfig={{
          name: 'user-name',
          detail: 'user-detail',
          initials: 'user-initials',
          imageSrc: 'user-src',
        }}
        variant="large"
        userActionConfig={userActionConfig}
      />,
    );

    expect(shallowComponent).toMatchSnapshot();
  });
});
