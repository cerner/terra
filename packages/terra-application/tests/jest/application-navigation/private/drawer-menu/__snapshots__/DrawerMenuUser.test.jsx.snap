// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerMenuUser should render default element 1`] = `
<div
  className="small-user-layout"
>
  <div
    className="avatar-container"
  >
    <div
      className="avatar-outline"
    />
    <div
      className="avatar-inner"
    >
      <Avatar
        alt="user-name"
        color="auto"
        image="user-src"
        initials="user-initials"
        isAriaHidden={true}
        isDeceased={false}
      />
    </div>
  </div>
  <div
    className="info-container"
  >
    <div
      aria-hidden={true}
      className="name"
    >
      user-name
    </div>
    <div
      className="detail"
    >
      user-detail
    </div>
    <button
      className="drawer-menu-action-button"
      data-focus-styles-enabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    >
      Edit Photo
    </button>
  </div>
</div>
`;

exports[`DrawerMenuUser should render with large variant 1`] = `
<div
  className="large-user-layout"
>
  <div
    className="avatar-container"
  >
    <div
      className="avatar-outline"
    />
    <div
      className="avatar-inner"
    >
      <Avatar
        alt="user-name"
        color="auto"
        image="user-src"
        initials="user-initials"
        isAriaHidden={true}
        isDeceased={false}
      />
    </div>
  </div>
  <div
    className="info-container"
  >
    <div
      aria-hidden={true}
      className="name"
    >
      user-name
    </div>
    <div
      className="detail"
    >
      user-detail
    </div>
    <button
      className="drawer-menu-action-button"
      data-focus-styles-enabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    >
      Edit Photo
    </button>
  </div>
</div>
`;

exports[`DrawerMenuUser should render with small variant 1`] = `
<div
  className="small-user-layout"
>
  <div
    className="avatar-container"
  >
    <div
      className="avatar-outline"
    />
    <div
      className="avatar-inner"
    >
      <Avatar
        alt="user-name"
        color="auto"
        image="user-src"
        initials="user-initials"
        isAriaHidden={true}
        isDeceased={false}
      />
    </div>
  </div>
  <div
    className="info-container"
  >
    <div
      aria-hidden={true}
      className="name"
    >
      user-name
    </div>
    <div
      className="detail"
    >
      user-detail
    </div>
    <button
      className="drawer-menu-action-button"
      data-focus-styles-enabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
    >
      Edit Photo
    </button>
  </div>
</div>
`;
